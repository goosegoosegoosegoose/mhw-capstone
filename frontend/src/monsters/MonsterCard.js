import React from "react";
import { Card } from "react-bootstrap";
import { useNavigate } from "react-router-dom";
import ToggleButton from "../common/ToggleButton";


const MonsterCard = ({id, name, icon, add, remove}) => {
  const nav = useNavigate();

  const handleClick = () => {
    nav(`/monsters/${id}`)
  }

  return(
    <Card className="my-3 mx-3 col-sm-3" style={{cursor: "pointer"}} bg="dark">
      <div className="row justify-content-center">
        <div className="col-sm-8 pt-2 mt-2">
          {icon ? <Card.Img style={{cursor: "pointer"}} onClick={handleClick} variant="top" src={icon} /> : <></>}
        </div>
      </div>
      <Card.Body className="d-flex flex-column">
        <Card.Title onClick={handleClick}>{name}</Card.Title>
        <ToggleButton id={Number(id)} type="monsters" spacing="my-1 mx-5 mt-auto" add={add} remove={remove} label1="Hunted" label2="Hunt"/>
      </Card.Body>
    </Card>
  )
}

export default MonsterCard;