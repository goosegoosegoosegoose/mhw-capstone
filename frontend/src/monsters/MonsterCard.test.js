import { render } from '@testing-library/react';
import MonsterCard from "./MonsterCard";
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <MonsterCard />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <MonsterCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});