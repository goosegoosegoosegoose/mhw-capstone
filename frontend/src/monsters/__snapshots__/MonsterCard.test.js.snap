// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3 mx-3 col-sm-3 card bg-dark"
    style="cursor: pointer;"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-sm-8 pt-2 mt-2"
      />
    </div>
    <div
      class="d-flex flex-column card-body"
    >
      <div
        class="card-title h5"
      />
      <button
        class="my-1 mx-5 mt-auto btn btn-danger btn-sm"
        disabled=""
        type="button"
      >
        Hunt
      </button>
    </div>
  </div>
</DocumentFragment>
`;
